### ISSUE TITLE
What features do you most want us to add ?

### ISSUE THREAD

**Issue by ResearcherXman** [InstantID/InstantID#56]
We are small team and cannot all feature requests at once. Thus, please tell us what is your favorite feature you wanna us add. Comment under this post. Meanwhile, we encourage all kinds of PR (fixing bugs, adding new features, etc.) from the community, if you want to take on a feature, please open a new issue naming "Support for xxx" and we will assign you. Thanks for your interests, love you all ❤️.

**Comment by fofr** [comment]
I'd love to see implementations of each of the examples in the paper.

**Comment by tgohblio** [comment]
**SD 1.5** support. 

Opens up whole new possibilities with LORAs, **AnimateDiff** (bcoz it's SD 1.5 only), community-trained SD1.5 models.
_Hint:_ this will drive more Github ⭐️

**Comment by nosiu** [comment]
> Not sure if its already there, but multiple reference images? I could not find a way to select more than one image in the sample code provided. Topic #33 mentioned use of multiple reference images for better results.
> 
> Also is it possible to do some "adetailer magic" like upscaling the part of the image that contains the face, apply hires face-swap and then resize down again to the original image to get a detailed face when not covering the whole image?
> 
> I am still new to this, so ignore anything i have misunderstood.

I'm no expert but this should work
`image = pipe(
            ... 
            image_embeds=(face_emb1 + face_emb2 + face_emb3 + face_emb4)/4,
            ...)
 `

**Comment by johndpope** [comment]
Downstream has some nice illustrations- https://github.com/ZHO-ZHO-ZHO/ComfyUI-InstantID - 
I played around with thie workflows - but couldn’t get results that I was after. It seems all the examples are based off text concatenation. - somehow interrogating image for words to add to prompt
Said another way - what I want is - given an image - can instantid make new images that look like it? Or is this not a suitable use case?
I realise t2i adapter colour will best  fit here - but then how does instantid shine?  Can you create something that looks like this?
 
![IMG_9879](https://github.com/InstantID/InstantID/assets/289994/16c9f088-a382-4f61-959c-3d9c756b1387)

**Comment by bent1e** [comment]
Adapt with IP Adapter to control the style of images by inputting ref instead of text prompt.

**Comment by johndpope** [comment]
Thanks @bent1e  - looks like the ref image has been updated. testing now
![Screenshot from 2024-01-25 13-46-02](https://github.com/InstantID/InstantID/assets/289994/9902d975-149d-4626-b5c7-42ba36999ada)

**Comment by marsplant** [comment]
Face swapper  is wanted!
<img width="1425" alt="compare-b" src="https://github.com/InstantID/InstantID/assets/78839542/4accf474-c4e0-4898-bf5f-9547fef5e20f">

**Comment by Kuvshin8** [comment]
Samplers? Sde/2M Karras or Euler A? I don't find any mentions about it

P.S. Thanks for your models

**Comment by brausesa** [comment]
Please make it possible for the local version to specify long tasks for generating images.... For example, so that I could specify one reference image of a face, plus specify several images of poses, plus, for example, select twenty different styles (prompts) and the number of variants (generations) for each style... and after pressing the "submit" button, I could go to sleep... And in the morning look ouput-folder with a hundred results :)

Thanks! You have made an incredible product!

**Comment by min-star** [comment]
> Face swapper is wanted! <img alt="compare-b" width="1425" src="https://private-user-images.githubusercontent.com/78839542/299566036-4accf474-c4e0-4898-bf5f-9547fef5e20f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MTcxMDIsIm5iZiI6MTcwNjYxNjgwMiwicGF0aCI6Ii83ODgzOTU0Mi8yOTk1NjYwMzYtNGFjY2Y0NzQtYzRlMC00ODk4LWJmNWYtOTU0N2ZlZjVlMjBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMwVDEyMTMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY3ZDI0OTMxN2U1OTIzYzdhN2JhZjgyNzU3YzcyZDYzYzJmNmJlYmVlYzA3ZWFiMTAyYWNlNTVhNTViMzc4OTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._lk1bcFPbPOph1EAuhsCpuG_v1va8HnCmOA9TF4V6sg">

InstantID is not only swapper face,did you use it with other ways

**Comment by ameen-roayan** [comment]
a simple comfy ui implementation would be great.

**Comment by johndpope** [comment]
Faceswapping is here - https://github.com/InstantID/InstantID/pull/89

**Comment by sbyebss** [comment]
Although it's already mentioned [here](https://github.com/InstantID/InstantID/issues/81), I hope Multi-ID can be added

**Comment by cqflgl** [comment]
项目很惊艳，非常感谢开源！
SD1.5版本的生态是最好的，如果能支持sd1.5，相信用户数和github星数都会上一个大台阶。

**Comment by cubiq** [comment]
Multi-ID is kinda possible already (in comfyui) even though a bit complicated at the moment, I'll see if I can find an easier way to do it https://github.com/InstantID/InstantID/issues/81#issuecomment-1937460812

**Comment by johndpope** [comment]
Im interested to use this technology with music. there was some prior work done with riffusion - but lack luster model support - this could with the help of a new model - reopen box to create styles of music from spectrograms + prompts.

**Comment by JohnCarpenter2** [comment]
Provide the official prompts for generating images in different styles.

**Comment by MartialWrapperr** [comment]
Full body of the subject instead of portrait mode.

**Comment by andymacool** [comment]
Please support other base models : )

**Comment by ynie** [comment]
Please train a model based on non-insight face weight

**Comment by JohnCarpenter2** [comment]
> Please train a model based on non-insight face weight

I think it is a good idea. Otherwise, provide the quick method to fine-tune the pre-training weights of InstantID on other face recognition models.

**Comment by lilminthehoodieboi** [comment]
sd 1.5 support please

**Comment by elismasilva** [comment]
could be possible we add mask to a face without use face analysis, then you could use this with instant style.

**Comment by leafiy** [comment]
full body with small head please！！
emotion control please！！

**Comment by jordancoult** [comment]
I don't know if this is possible but, it would be insane.
InstantID for flux.

**Comment by appm111** [comment]
Add text support soon as possible 👍

**Comment by boco927** [comment]
really need multi-id function realized by code rather than comfyui because many machines doesn't support GUI.

**Comment by Neural-type** [comment]
We need flux support

**Comment by MorningStarJ** [comment]
Supporting flux would be an exciting new feature.

**Comment by platform-kit** [comment]
Flux 100%

**Comment by AoeSyL** [comment]
flux！

**Comment by africa1207** [comment]
Supporting flux!

**Comment by Ivang71** [comment]
flux support would be awesome
